design/login.md

```diff
@@ -116,3 +117,46 @@ Each login provider has one of these 5 login methods associated to it (`client-m
     Authorization header in order to verify the token.
 
 > `start_flow_url` can either be an absolute URL or a cluster relative path.
+
+## Local web server
+
+When initiating a flow for the `dcos-oidc-auth0` login provider ID, the CLI will spin-up
+a local web server on a free port. This is done by using port `0`.
+
+The CLI then opens the user browser (using `xdg-open <url>` on Linux, `open <url>` on macOS,
+`rundll32 url.dll,FileProtocolHandler <url>` on Windows) at the `start_flow_url` with 2 extra query parameters:
+
+- `redirect_uri`: refers to the URL where the local web server is listening.
+
+- `dcos_cli_csrf_token`: contains a 32 bytes base64 token created with the Go `crypto/rand` stdlib package.
+
+In case the browser didn't open (eg. SSH session on a remote machine), the user also sees the following
+message:
+
+``` console
+If your browser didn't open, please follow this link:
+
+    http://my-cluster.example.com/login?redirect_uri=http://localhost:8080&dcos_cli_csrf_token=g6qAFJeHUz3OViQHPPnwkCSjo3BVZSn4QiqmtrlqElo=
+```
+
+On successful login, our [Auth0 universal login page](https://github.com/mesosphere/auth0-ui) is
+configured to make a `GET` request to the `redirect_uri`, once it has validated that it refers to a `localhost` URL, with 2 parameters:
+
+- `token` contains the login token.
+
+- `csrf` contains the CSRF token that was initially set in `dcos_cli_csrf_token`.
+
+For example:
+
+    http://localhost:8080?token=myLoginToken123&csrf=g6qAFJeHUz3OViQHPPnwkCSjo3BVZSn4QiqmtrlqElo=
+
+The Auth0 universal login page uses [fetch()](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch)
+in order to perform the request. As this is a cross-domain request, the local web server
+is configured to support CORS and will only accept requests from the `https://dcos.auth0.com` origin.
+
+Once the request is done, the local web server can retrieve the token and continue the login flow.
```

Comment:
Add that the web server must check that the CSRF token returned in the redirect URI matches the one sent.